manifests/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -144,6 +151,12 @@ spec:
 {{- if .Values.pilot.traceSampling }}
           - name: PILOT_TRACE_SAMPLING
             value: "{{ .Values.pilot.traceSampling }}"
+{{- end }}
+{{- if .Values.global.revision }}
+          - name: REVISION
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.labels['istio.io/rev']
```

Comment:
The value of `istio.io/rev` should be `.Values.global.revision`. You could set the envvar directly in the deployment. That would get around the immutable deployment label selector for 1.5 I think? We could add an annotation to the pod for purely informational purposes, but that would only exist for Istio < 1.6.

Reply:
I think we want it to have a different selector intentionally though, right? otherwise the two are coupled. I need to look into what happens with selectors overlapping on deployments, I don't quite understand.

But I also think we want a label - we might actually want it on every single resource, so we can do `kubectl apply --prune -listio.io/rev=foo` type things? Otherwise it will be hard to clean up

Reply:
A revision label on the deployment to match the associated resources is desirable and something I think we can make work in 1.6. IIUC, that requires proper multi-control plane support (e.g. istiod-<revision>)?

My understanding of this PR is to make istiod (1.5) forward compatible with 1.6 and honor the `istio.io/revision` label on resources. It would be great if we could sync the resource revision labels with the deployment revision label but that didn't seem possible and I don't think is strictly required. As a short term solution you could hardwire the `REVISION` envvar so that config filtering doesn't depend on the deployment's revision label, e.g.

```yaml
{{- if .Values.global.revision }}
          - name: REVISION
            valueFrom:
              fieldRef:
                fieldPath: {{ Values.global.revision }}
{{- end }}
```